pkg/ipamd/ipamd_test.go

```diff
@@ -1850,6 +1850,26 @@ func TestIPAMContext_askForTrunkENIIfNeeded(t *testing.T) {
 	defer m.ctrl.Finish()
 	ctx := context.Background()
 
+	fakeRecorder := events.NewFakeRecorder(3)
+	k8sSchema := runtime.NewScheme()
+	clientgoscheme.AddToScheme(k8sSchema)
+
+	mockEventRecorder := &eventrecorder.EventRecorder{
+		Recorder:        fakeRecorder,
+		RawK8SClient:    testclient.NewClientBuilder().WithScheme(k8sSchema).Build(),
+		CachedK8SClient: testclient.NewClientBuilder().WithScheme(k8sSchema).Build(),
+	}
+	eventrecorder.MyNodeName = myNodeName
```

Comment:
nit: this is strange to read, can you add a comment that this is setting global var in `eventrecorder` package